app/models/root-store/root-store.ts

```diff
@@ -90,8 +91,10 @@ export const RootStoreModel = types
       self.isShowUnauthenticatedAlert = false
       self.navigationStore.navigateTo("Auth")
       yield self.userStore.logout()
-      self.readerStore.clearAllLists()
-      self.chainStore = undefined
+      self.chainStore.reset()
+      applySnapshot(self.readerStore, {})
+      applySnapshot(self.statisticsRewardedStore, {})
+      applySnapshot(self.statisticsSupportedStore, {})
```

Comment:
Should every store have a destructor method like `.reset()`, then we just call them all here without caring details for each store?

Reply:
Updated